src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= c + b || b >= a + c || c >= b + a) {
+      throw new Error(
+        'the longest side is more than the sum of two other sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const q = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(q * (q - this.a) * (q - this.b) * (q - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape = Shape.Circle;
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('radius is less than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('width or height are less than 0');
+    }
+  }
+
+  getArea(): number {
```

Comment:
The `getArea` method should not accept any parameters. It should use the `width` and `height` fields of the object.